src/Access/SettingsAuthResponseParser.cpp

```diff
@@ -28,7 +28,7 @@ SettingsAuthResponseParser::parse(const Poco::Net::HTTPResponse & response, std:
     try
     {
         Poco::Dynamic::Var json = parser.parse(*body_stream);
-        Poco::JSON::Object::Ptr obj = json.extract<Poco::JSON::Object::Ptr>();
+        const Poco::JSON::Object::Ptr & obj = json.extract<Poco::JSON::Object::Ptr>();
```

Comment:
This is `clang19` related change, there are some more in this PR. @rschu1ze do you think I may keep them in this PR, or should we split it?

Reply:
Totally fine to keep them in this PR as long as they don't break compilation with earlier Clang versions (the lowest possible Clang version is defined by `CLANG_MINIMUM_VERSION` in cmake/tools.cmake). In this case, I don't see any backward compatibility issues.